extras/bindata.bzl

```diff
@@ -43,7 +43,10 @@ def _bindata_impl(ctx):
         arguments.add_all(["-modtime", "0"])
     if ctx.attr.extra_args:
         arguments.add_all(ctx.attr.extra_args)
-    arguments.add_all(ctx.files.srcs)
+    srcs = [f.path for f in ctx.files.srcs]
+    if ctx.attr.experimental_strip_external and [f for f in srcs if f.startswith("external/")]:
+        arguments.add_all(["-prefix", "/".join([ctx.label.workspace_root, ctx.label.package])])
```

Comment:
This all can be done lazily to avoid constructing `srcs` in the analyzis phase. Something like:
```bzl
def _maybe_add_prefix(t):
    if [f for f in t.srcs if f.path.startswith("external/")]:
        # use paths.join() from skylib to construct path
        return ["-prefix", paths.join(t.workspace_root, t.package)]
    else:
        return None
if ctx.attr.experimental_strip_external:
    arguments.add_all([struct(srcs = ctx.files.srcs, workspace_root = ctx.label.workspace_root, package = ctx.label.package)], map_each = _maybe_add_prefix)
arguments.add_all(ctx.files.srcs)
``

Reply:
This would make sense if were were accumulating a depset of sources over a potentially large tree, but I think it's overkill in this situation.